code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -10325,3 +10325,32 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 			new /obj/effect/alien/weeds(T)
 		if(!locate(/obj/effect/decal/cleanable/purpledrank) in T)
 			new /obj/effect/decal/cleanable/purpledrank(T)
+
+/datum/reagent/punctualite
+	name = "Punctualite"
+	id = PUNCTUALITE
+	description = "Nicknamed mad chemist's alarm clock. Explodes on the hour when being actively metabolized. "
+	reagent_state = REAGENT_STATE_LIQUID
+	color = "#8d8791" //rgb: 200, 165, 220
+	custom_metabolism = 0 //Wouldn't be much fun if it all got metabolized beforehand
+	var/currentHour = 0 //The hour it was introduced into the system so it doesn't blow right away
+
+/datum/reagent/punctualite/on_mob_life(var/mob/living/M)
+	if(..())
+		return 1
+	if(!currentHour)
+		currentHour = floor(world.time / 36000) + 1
+	if(prob(5) && prob(5)) //0.25% chance per tick
```

Comment:
Tick... tock... would be more grammatically pleasing.

Reply:
I also don't like how it's a visible message. What's actually happening? Do we hear the sound?

Reply:
yeah uh did you test this? visible messages and playsounds from things that aren't directly on a turf don't really work

Reply:
I mean, it wasn't really my idea. I don't think it needs it to begin with but people wanted a visible message.

Reply:
This is formatted strangely as Shifty said. If it's a notice message, it should be something you notice, as in:
`You hear a ticking sound...`
or
`You hear a ticking sound coming from \the [M].`
Should this be visible to deaf players? Probably not.

Reply:
I think I got it right that time